### DISCUSSION TITLE
How do I build wezterm for Windows?

### DISCUSSION THREAD

**Discussion by cleemesser** [wezterm/wezterm#5521] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### Which Wayland compositor or X11 Window manager(s) are you using?

NA

### WezTerm version

wezterm-20240203-110809-5046fc22

### Ask your question!

Hi, I am a rust newbie and I have not been developing on Windows 11 a lot recently. except for in WSL2/ubuntu 22.04. I do most development on unix/linux variants and building wezterm from source is easy for me there.

However, I do not see how to build the native Windows wezterm. I've downloaded the source, but get-deps does not seem to cover windows and relies on sh to run. 

It is not clear how I am supposed to get the library dependencies on Windows.

I have, as a somewhat random attempt, tried using on the bash installed with Git for Windows. but that is not recognized by get-deps.

On Windows 11 rustup/rustc/cargo are all installed and working -- surprisingly well as I was able to repeat some of the simple tutorial projects for rust I did previously under linux on my windows machine without a problem.

**Comment by wez** [comment]
I thought we'd transferred this to the docs, but apparently not yet.
This comment has some info for you:
https://github.com/wez/wezterm/issues/3425#issuecomment-1493235496

**Reply by cleemesser** [reply]
Thank you for the guidance. It worked. The strawberry perl part was the trick. It was trying to use the perl that comes with Git for Windows.

**Reply by cleemesser** [reply]
I've created an answer in markdown below. Let me know if you want the file as a pull request, say a file called INSTALL-WINDOWS.md or if you want it somewhere else. Also, thank you for wezterm.

**Reply by cleemesser** [reply]
I put in a PR for your review which added to the docs

**Comment by cleemesser** [comment, marked as answer]
# How to build wezterm on windows, based upon #3425 (comment) 

1. install windows rust from [the rust website](https://www.rust-lang.org/tools/install) following their instructions.

- Note info on installing other tools for using rust on windows is covered at the microsoft [rust dev environment overview on windows](https://learn.microsoft.com/en-us/windows/dev-environment/rust/overview) page.

2. Install [Strawberry Perl](https://strawberryperl.com/)

- I downloaded the msi installer from the Strawberry Perl website installed,  accepting the defaults to install in c:\Strawberry
- I had to alter the path for the install to work (see below)

I opened up a new cmd.exe terminal in wezterm :-)

I checked that strawberry perl is now installed and it will be used
```
> where perl
C:\Program Files\Git\usr\bin\perl.exe
```

Uh oh, git for windows has its own version of perl. Can't use that. (Per Wez's insructions, only Strawberry Perl works. So, I set the PATH so that Strawberry perl is found first:

```
set PATH=c:\Strawberry\perl\bin;%PATH%

> where perl
c:\Strawberry\perl\bin\perl.exe
C:\Program Files\Git\usr\bin\perl.exe

> perl --version
This is perl 5, version 38, subversion 2 (v5.38.2) built for MSWin32-x64-multi-thread
```


3. clone and build 
```
git clone --depth=1 --branch=main --recursive https://github.com/wez/wezterm.git
cd wezterm
git submodule update --init --recursive
cargo build 
```
Yeah, now the openssl-sys build does not give the error!

4. finally I did a test run:
```
cargo run --bin wezterm
```

And it wezterm should pop up in a new window.


Previously, When I naively tried to do the rust build process I got this error during the build after a bunch of modules were compiled correctly:

```

  BEGIN failed--compilation aborted at ./Configure line 23. 
  thread 'main' panicked at C:\Users\clee\.cargo\registry\src\index.crates.io-6f17d22bba15001f\openssl-src-300.3.1+3.3.1\src\lib.rs:621:9:



  Error configuring OpenSSL build:
      Command: "perl" "./Configure" "--prefix=C:/Users/clee/code/wezterm/target/debug/build/openssl-sys-ec8b5d23e810c034/out/openssl-build/install" "--openssldir=SYS$MANAGER:[OPENSSL]" "no-dso" "no-shared" "no-ssl3" "no-tests" "no-comp" "no-zlib" "no-zlib-dynamic" "--libdir=lib" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "no-capieng" "VC-WIN64A"
      Exit status: exit code: 2
	  
```

**Comment by FrancescElies** [comment]
Anyone having this issue when building on windows?

```
  --- stderr
  ..\locale.c: 690: panic: (6578): Can't change locale for LC_CTYPE(2) from 'C' to 'English_Europe.65001'; errno=0
  thread 'main' panicked at C:\Users\FrancescElies\.cargo\registry\src\index.crates.io-6f17d22bba15001f\openssl-src-300.3.1+3.3.1\src\lib.rs:621:9:
```

I found the following [issue](https://github.com/StrawberryPerl/Perl-Dist-Strawberry/issues/135) but didn't manage to make the first proposed a workarund work.
I also checked how things are being built [in ci](https://github.com/wez/wezterm/blob/main/.github/workflows/gen_windows.yml) but didn't see anything special about locale.

This is definiterly a perl issue (not wezterm) but trying my chances here too.
![image](https://github.com/user-attachments/assets/d10082bd-05e2-4b1e-be01-f94d1ac0edec)

**Reply by FrancescElies** [reply]
As pointed out [here](https://github.com/StrawberryPerl/Perl-Dist-Strawberry/issues/135#issuecomment-2261523716) using 5.4 RC1 solves that issue, as soon it's released installing perl from the website should be fine too.
